x-pack/plugins/monitoring/server/lib/cluster/flag_supported_clusters.js

```diff
@@ -78,7 +78,7 @@ export function flagSupportedClusters(req, kbnIndexPattern) {
     // if multi cluster
     if (clusters.length > 1) {
       const basicLicenseCount = clusters.reduce((accumCount, cluster) => {
```

Comment:
Nit: could we update the variable name here, since it's no longer just about the license?

Reply:
Also, further below in this same function, could we update the log messages as well? I can't comment on those lines because they haven't been changed by this PR.

Reply:
Looks like the header comment on the `flagSupportedClusters` function should be updated as well.

Reply:
Looking at the logic of the "all basic licenses" block here:

```
      // if all basic licenses
      if (clusters.length === basicLicenseCount) {
        const kibanaUuid = config.get('server.uuid');
        return await findSupportedBasicLicenseCluster(req, clusters, kbnIndexPattern, kibanaUuid, serverLog);
      }
```

If I'm following this right, in a scenario where there's an unlinked cluster and a basic cluster (to which the current kibana is attached), only the basic cluster will be supported. But what we want is for the unlinked cluster to be supported as well, right?

Reply:
Similar to the previous comment in this thread, I'm not sure the "some basic licenses" block works as we want either, if I'm following the logic correctly.

Consider the scenario of 1 unlinked cluster, 1 basic cluster, and 1 trial cluster. `clusters.length` === 3 and `basicLicenseCount` === 2. So we end up in the "some basic licenses" code block, whose logic is thus:

```
      // if some non-basic licenses
      serverLog('Found some basic license clusters in monitoring data. Only non-basic will be supported.');
      clusters.forEach(cluster => {
        if (cluster.license && cluster.license.type !== 'basic') {
          cluster.isSupported = true;
        }
```

I'd expect `cluster.license` to be falsy for the unlinked cluster, and as a result this cluster's `isSupported` wouldn't be set to `true`.

Am I following this right?

Reply:
Just a suggestion: might it be cleaner to do something like this instead?

1. Find if there is an unlinked cluster in `clusters`. If so, set it's `isSupported` flag to `true`.
2. Create a new variable, `linkedClusters` which is `clusters` sans the unlinked one. And then use the same logic starting with the `// if multi clusters` line as today, but on `linkedClusters` instead of `clusters`.